conans/test/unittests/model/build_info/components_test.py

```diff
@@ -176,3 +180,237 @@ def test_deps_cpp_info_libs_release_debug(self):
+                    for lib in info.components[dep].libs:
+                        self.assertIn(lib, sorted_libs[num:])
+
+        info = CppInfo("")
+        info.components["6"].libs = ["lib6"]
+        info.components["6"].requires = ["::4", "::5"]
+        info.components["5"].libs = ["lib5"]
+        info.components["5"].requires = ["::2"]
+        info.components["4"].libs = ["lib4"]
+        info.components["4"].requires = ["1"]
+        info.components["3"].libs = ["lib3"]
+        info.components["3"].requires = ["::1"]
+        info.components["1"].libs = ["lib1"]
+        info.components["1"].requires = ["::2"]
+        info.components["2"].libs = ["lib2"]
+        info.components["2"].requires = []
+        dep_cpp_info = DepCppInfo(info)
+        _assert_link_order(dep_cpp_info.libs)
+        self.assertEqual(["lib6", "lib5", "lib4", "lib3", "lib1", "lib2"], dep_cpp_info.libs)
+        deps_cpp_info = DepsCppInfo()
+        deps_cpp_info.update(dep_cpp_info, "dep1")
+        self.assertEqual(["lib6", "lib5", "lib4", "lib3", "lib1", "lib2"],
+                         deps_cpp_info.libs)
+
+        info = CppInfo("")
+        info.components["K"].libs = ["libK"]
+        info.components["K"].requires = ["::G", "::H"]
+        info.components["J"].libs = ["libJ"]
+        info.components["J"].requires = ["::F"]
+        info.components["G"].libs = ["libG"]
+        info.components["G"].requires = ["::F"]
+        info.components["H"].libs = ["libH"]
+        info.components["H"].requires = ["::F", "::E"]
+        info.components["L"].libs = ["libL"]
+        info.components["L"].requires = ["::I"]
+        info.components["F"].libs = ["libF"]
+        info.components["F"].requires = ["::C", "::D"]
+        info.components["I"].libs = ["libI"]
+        info.components["I"].requires = ["::E"]
+        info.components["C"].libs = ["libC"]
+        info.components["C"].requires = ["::A"]
+        info.components["D"].libs = ["libD"]
+        info.components["D"].requires = ["::A"]
+        info.components["E"].libs = ["libE"]
+        info.components["E"].requires = ["::A", "::B"]
+        info.components["A"].libs = ["libA"]
+        info.components["A"].requires = []
+        info.components["B"].libs = ["libB"]
+        info.components["B"].requires = []
+        dep_cpp_info = DepCppInfo(info)
+        _assert_link_order(dep_cpp_info.libs)
+        self.assertEqual(["libK", "libJ", "libG", "libH", "libL", "libF", "libI", "libC", "libD",
+                          "libE", "libA", "libB"], dep_cpp_info.libs)
+        deps_cpp_info.update(dep_cpp_info, "dep2")
+        self.assertEqual(["lib6", "lib5", "lib4", "lib3", "lib1", "lib2","libK", "libJ", "libG",
+                          "libH", "libL", "libF", "libI", "libC", "libD", "libE", "libA", "libB"],
+                         deps_cpp_info.libs)
+
+    def cppinfo_inexistent_component_dep_test(self):
+        info = CppInfo(None)
+        info.components["LIB1"].requires = ["::LIB2"]
```

Comment:
I thought we were ok with the syntax:
- ``requires = ["Lib2"]`` for intra-package dependencies
- ``requires = ["pkg::other"]`` for dependencies to other package components

Is there something new that we didn't realize that disables this possibility?